ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx

```diff
@@ -0,0 +1,253 @@
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const { sortOption, getSortParams } = useURLSort({
+        sortFields,
+        defaultSortOption,
+    });
+
+    const listQuery = useCallback(
+        () =>
+            getSingleClusterResultsByScanConfig(
+                clusterId,
+                scanName,
+                searchFilter,
+                sortOption,
+                page - 1,
+                perPage
+            ),
+        [clusterId, scanName, searchFilter, sortOption, page, perPage]
+    );
+    const { data: scanResults, loading: isLoading, error } = useRestQuery(listQuery);
+
+    const countQuery = useCallback(
+        () => getSingleClusterResultsByScanConfigCount(clusterId, scanName, searchFilter),
+        [clusterId, scanName, searchFilter]
+    );
+    const { data: scanResultsCount } = useRestQuery(countQuery);
+
+    function getStatusByClusterId(clusters: ClusterCheckStatus[]): ComplianceCheckStatus | null {
+        const cluster = clusters.find((cluster) => cluster.cluster.clusterId === clusterId);
+        return cluster ? cluster.status : null;
+    }
+
+    function onChangeSearchFilter(newFilter: SearchFilter) {
+        setSearchFilter(newFilter);
+    }
+
+    function onSearchInputChange(_event, value) {
+        setSearchCheckValue(value);
+    }
+
+    const handleCheckInputSearch = () => {
+        const newFilter = {
+            ...searchFilter,
+            'Compliance Check Name': searchCheckValue,
+        };
+        setSearchFilter(newFilter);
+    };
+
+    const handleCheckInputClear = () => {
+        const newFilters = omit(searchFilter, 'Compliance Check Name');
+        setSearchCheckValue('');
+        setSearchFilter(newFilters);
+    };
+
+    function onSelect(type: 'Compliance Check Status', checked: boolean, selection: string) {
+        const selectedSearchFilter = searchValueAsArray(searchFilter[type]);
+        onChangeSearchFilter({
+            ...searchFilter,
+            [type]: checked
+                ? [...selectedSearchFilter, selection]
+                : selectedSearchFilter.filter((value) => value !== selection),
+        });
+    }
+
+    const renderTableContent = () => {
+        return scanResults?.checkResults.map(({ checkName, rationale, clusters }) => {
+            const scanConfigUrl = generatePath(complianceEnhancedScanConfigDetailPath, {
+                scanConfigId: checkName,
+            });
+
+            const status = getStatusByClusterId(clusters);
+            const statusObj = status ? getClusterResultsStatusObject(status) : null;
+
+            return (
+                <Tr key={checkName}>
+                    <Td modifier="truncate">
+                        <>
+                            <Link to={scanConfigUrl}>{checkName}</Link>
+                            <br />
+                            <small>{rationale}</small>
```

Comment:
for a TODO: It seems like there is a PatternFly component for this look, with main info larger on top, and further info smaller on the bottom.

Reply:
Hmm any idea where they have that? I couldn't find a devoted component so I ended up mimicking a pattern in one of the table examples [here](https://v4-archive.patternfly.org/v4/components/table#composable-nested-column-headers). Although the example is just noting the striped rows and doesn't appear to be intentionally included in the component. They also have [this](https://v4-archive.patternfly.org/v4/components/table#composable-nested-column-headers) which includes faded text but not small.